dev/PackageManager/API/PackageManager.idl

```diff
@@ -0,0 +1,383 @@
+
+        static PackageVolume FindPackageVolumeByName(String name);
+    };
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageSetItem
+    {
+        PackageSetItem();
+
+        // Unique id
+        String Id;  //required
+
+        // Package criteria to identify if a matching package is ready for use and to use at runtime
+        String PackageFamilyName;                           //required
+        Windows.ApplicationModel.PackageVersion MinVersion; //required
+//TODO        Microsoft.Windows.ApplicationModel.ProcessorArchitectures ProcessorArchitectureFilter;
+
+        // Source where a package can be retrieved from if/when needed for install, remediation, etc
+        Windows.Foundation.Uri PackageUri;
+
+        // ??? Additional info potentially needed for deployment operations
+        //TODO see AddPackageOptions
+
+        // Additional info for runtime use
+        //TODO see CreatePackageDependencyOptions
+        //TODO see AddPackageDependencyOptions
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageSet
+    {
+        PackageSet();
+
+        // Unique id
+        String Id;
+
+        // Package(s) in the set
+        IVector<PackageSetItem> PackageSetItems { get; };
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass FindPackageOptions
+    {
+        FindPackageOptions();
+
+        //TODO
+        Boolean TODO_MustHaveAtLeastOneInterfaceOrStaticFactoryPlaceholder;
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageSetManager
+    {
+        static PackageSetManager GetDefault();
+
+        void Add(PackageSet packageSet);  		                // Create/Write/Save
+        void Update(PackageSet packageSet);  		            // Update/Write/Save
+        void Remove(String packageSetId);		                // Delete
+        PackageSet Get(String packageSetId);		            // Read/Load
+        IVector<PackageSet> Find(FindPackageOptions options);   // Enumerate
+    };
+
+    [contract(PackageDeploymentContract, 1)]
+    enum PackageDeploymentStatus
+    {
+        InProgress = 0,         // The request is in progress
+        CompletedSuccess = 1,   // The request completed successfully
+        CompletedFailure = 2,   // The request failed with some critical internal error. //TODO:Check ExtendedError for HRESULT
+    };
+
+    [contract(PackageDeploymentContract, 1)]
+    enum PackageDeploymentProgressStatus
+    {
+        Queued = 0,             // The request is queued
+        InProgress = 1,         // The request is in progress
+        CompletedSuccess = 2,   // The request completed successfully
+        CompletedFailure = 3,   // The request failed with some critical internal error. //TODO:Check ExtendedError for HRESULT
+    };
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageDeploymentProgress
+    {
+        PackageDeploymentProgressStatus status;
+        UInt32 percentage;
```

Comment:
This was an error on our part. We should have made this a `Single` since there are multi-gb packages in the world.

Reply:
Good idea, though Double seems more right than Single here. Using Tempo I see many progress properties are uint, some are double (e.g. MediaElement's.DownloadProgress) and I only see one using float (AnimationController), and notably PackageCatalog uses Double e.g. [`PackageInstallingEventArgs.Progress` property](https://learn.microsoft.com/uwp/api/windows.applicationmodel.packageinstallingeventargs.progress?view=winrt-22621#windows-applicationmodel-packageinstallingeventargs-progress)